rocks-lib/src/rockspec/build/builtin.rs

```diff
@@ -1,23 +1,30 @@
 use std::{collections::HashMap, path::PathBuf};
 
+use eyre::{OptionExt, Result};
+use itertools::Itertools;
 use serde::{de, Deserialize, Deserializer};
+use walkdir::WalkDir;
+
+use crate::{rockspec::Rockspec, tree::TreeLayout};
+
+use super::Build;
 
 #[derive(Debug, PartialEq, Deserialize, Default, Clone)]
 pub struct BuiltinBuildSpec {
     /// Keys are module names in the format normally used by the `require()` function
-    pub modules: HashMap<String, ModulesSpec>,
+    pub modules: HashMap<String, ModuleType>,
 }
 
 #[derive(Debug, PartialEq, Clone)]
-pub enum ModulesSpec {
+pub enum ModuleType {
```

Comment:
thought: I find the word "type" here a bit confusing, given that rust has a nominal type system. 
The type here is `ModuleSpec` (or `ModuleType`), and each of the variants is a constructor.

That's why I chose `Spec`.

Reply:
To me, a spec is for structures, not for variations of a single bit of data (like the enum in this case).

It's not a convention to call types `xyzType` in Rust either, I just feel like it's much easier to grasp the intent at a glance. It's the "type" of "module" to be used :p

This is all bikeshedding in the end, so I don't mind whether it's called this or that, I just find it a more deliberate naming scheme.

Reply:
Fair enough, I'll move it back to `ModulesSpec` for consistency's sake :)